pandas/tests/test_index.py

```diff
@@ -60,6 +60,16 @@ def test_shift(self):
         self.assertRaises(NotImplementedError, idx.shift, 1)
         self.assertRaises(NotImplementedError, idx.shift, 1, 2)
 
+    def test_create_index_existing_name(self):
+
+        # GH11193, when an existing index is passed, and a new name is not specified, the new index should inherit the
+        # previous object name
+        idx = self.create_index()
+        idx.name = "foo"
+        result = pd.Index(idx)
+        self.assertTrue("'foo'" in str(result))
```

Comment:
just construct the expected index and compare
```
expected = self.create_index()
expected.name = 'foo'
tm.assert_index_equal(result, expected)
```

also check with a new passed name